modules/primo/src/Primo/Ting/Object.php

```diff
@@ -162,113 +181,197 @@ public function getPublisher() {
    */
-  public function getSeriesDescription() {
-    // TODO: Implement getSeriesDescription() method.
+  public function getURI() {
+    // TODO: Implement getURI() method.
   }
 
   /**
    * @inheritDoc
    */
-  public function getSource() {
-    $source = $this->document->getLocalDisplayField(6);
-    // Source texts may contain the text 'Á frummáli: ' meaning 'In the original
-    // language'. However such a prefix will be added by the calling code so
-    // we remove it here.
-    // Note that this is strictly tied to how Primo is used by Icelandic
-    // libraries.
-    return str_replace('Á frummáli: ', '', $source);
+  public function getType() {
+    return $this->document->getType();
   }
 
   /**
    * @inheritDoc
    */
-  public function getSpatial() {
-    // TODO: Implement getSpatial() method.
+  public function getYear() {
+    return $this->document->getYear();
   }
 
+  // Below are parts of the TingObjectInterface which the Primo modules
+  // currently do not support.
+  // Note that this does not necessarily mean that the information is not
+  // available from Primo. It is just not implemented at the moment. Please
+  // check each getter for any additional information.
+
   /**
    * @inheritDoc
    */
-  public function getSpoken() {
-    // TODO: Implement getSpoken() method.
+  public function getAbstract() {
+    // Return nothing. We do not support abstracts.
   }
 
   /**
    * @inheritDoc
    */
-  public function getSubTitles() {
-    // TODO: Implement getSubTitles() method.
+  public function getClassification() {
+    // Return nothing. We do not support classification.
   }
 
   /**
    * @inheritDoc
    */
-  public function getTracks() {
-    // TODO: Implement getTracks() method.
+  public function getFormat() {
+    // Return nothing. The contents of this field is partly duplicated by
+    // getExtent().
   }
 
   /**
    * @inheritDoc
    */
-  public function getURI() {
-    // TODO: Implement getURI() method.
+  public function isPartOf() {
+    // Return nothing. IsPartOf is not supported at the moment.
   }
 
   /**
    * @inheritDoc
    */
-  public function getVersion() {
-    // Do nothing. Primo does not distinguish between difference versions of
-    // the same object.
+  public function getMusician() {
+    // Return nothing. We do not support musician information.
   }
 
   /**
    * @inheritDoc
    */
-  public function isPartOf() {
-    // TODO: Implement isPartOf() method.
+  public function getPegi() {
+    // Return nothing. We do not support PEGI information.
   }
 
   /**
    * @inheritDoc
    */
-  public function getType() {
-    return $this->document->getFormat();
+  public function getReferenced() {
+    // Return nothing. We do not support reference information.
   }
 
   /**
    * @inheritDoc
    */
-  public function getYear() {
-    return $this->document->getYear();
+  public function getRelations() {
+    // Return nothing. We do not support relations at the moment.
```

Comment:
Buuuut we still need to return something right? We could have a base class (or trait ?) to make this easier for providers ....

Reply:
:+1: Fixed